packages/suite-build/configs/base.webpack.config.ts

```diff
@@ -0,0 +1,148 @@
+import webpack from 'webpack';
+import { BundleAnalyzerPlugin } from 'webpack-bundle-analyzer';
+
+import alias from '../utils/alias';
+import { assetPrefix, isDev, isAnalyzing, environment } from '../utils/env';
+import { generateCacheGroups } from '../utils/suite';
+
+const suiteCacheGroups = generateCacheGroups([
+    'components',
+    'views',
+    'actions',
+    'reducers',
+    'utils',
+    'hooks',
+    'support',
+    'middlewares',
+    'constants',
+    'services',
+    'config',
+    'storage',
+]);
+const config: webpack.Configuration = {
+    mode: environment,
+    target: isDev ? 'web' : 'browserslist',
+    devtool: isDev ? 'eval-source-map' : false,
+    devServer: {
+        port: 3000,
+        hot: isDev, // HMR
+    },
+    output: {
+        publicPath: `${assetPrefix}/`,
+        filename: isDev ? 'js/[name].js' : 'js/[name].[contenthash:8].js',
+        chunkFilename: isDev ? 'js/[id].js' : 'js/[id].[contenthash:8].js',
+        assetModuleFilename: `assets/[hash][ext][query]`,
+        pathinfo: false,
+    },
+    resolve: {
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        modules: ['node_modules'],
+        alias,
+        fallback: {
+            // Polyfills API for NodeJS libraries in the browser
+            crypto: require.resolve('crypto-browserify'),
+            os: require.resolve('os-browserify/browser'),
+            path: require.resolve('path-browserify'),
+            stream: require.resolve('stream-browserify'),
+            buffer: require.resolve('buffer'),
+            process: require.resolve('process'),
+            // For Google OAuth library to work
+            child_process: false,
+            fs: false,
+            net: false,
+            tls: false,
+        },
+    },
+    optimization: {
+        splitChunks: {
+            cacheGroups: {
+                react: {
+                    chunks: 'initial',
+                    name: 'react',
+                    test: /[\\/]node_modules[\\/]react/,
+                },
+                vendors: {
+                    chunks: 'initial',
+                    name: 'vendors',
+                    test: /[\\/]node_modules[\\/](?!react)/,
+                },
+                components: {
+                    chunks: 'initial',
+                    name: 'components',
+                    test: /[\\/]packages[\\/]components[\\/]/,
+                },
+                ...suiteCacheGroups,
+            },
+        },
```

Comment:
The webpack docs [discourage](https://webpack.js.org/plugins/split-chunks-plugin/#configuration) custom `splitChunks` config. Could we maybe add a comment explaining why we're having this custom config?

Reply:
It's common practice. This is solving the issue of having too many or too heavy chunks.